src/Tools/dotnet-monitor/Egress/S3/S3StorageEgressProvider.cs

```diff
@@ -0,0 +1,258 @@
+using Amazon.Runtime;
+using Amazon.Runtime.CredentialManagement;
+using Amazon.S3;
+using Amazon.S3.Model;
+using Amazon.S3.Transfer;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Security;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.Egress.S3
+{
+    /// <summary>
+    /// Egress provider for egressing stream data to the s3 storage.
+    /// </summary>
+    internal class S3StorageEgressProvider : EgressProvider<S3StorageEgressProviderOptions>
```

Comment:
I would like to see some way of unit testing this provider without actually using a hosted S3 account. Here's how we are testing the Azure blob egress provider: https://github.com/dotnet/dotnet-monitor/blob/main/src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/Egress/AzureBlob/AzureBlobEgressProviderTests.cs